### PR TITLE
[csrng/dv] Update testplan/checklist, add coverage

### PR DESCRIPTION
**Pull request by senelson7**
Updates for CSRNG V2 review.

Signed-off-by: Steve Nelson <steve.nelson@wdc.com>

### CODE REVIEW

**Review by mwbranstad — APPROVED** [review]
LGTM.

File: hw/ip/csrng/data/csrng_testplan.hjson [hunk 1]
```diff
@@ -59,17 +61,19 @@
             Verify genbits generated as predicted.
             Verify fips bit is passed through properly.
             Verify ability to access registers based on otp, enables.
+            Verify AES_HALT.
+            * Verify continuous/non-continuous valid.
```
Comment (weicaiyang) [inline hw/ip/csrng/data/csrng_testplan.hjson:65]: nit: why there is a `*`

File: hw/ip/csrng/data/csrng_testplan.hjson [hunk 2]
```diff
@@ -83,6 +87,26 @@
...
+            '''
+    }
+    {
+      name: csrng_cmds_cg
+      desc: '''
+            Covers that all csrng commands and variations have been tested for all apps.
+            Individual commands and command options that will be covered include:
+            - app
+            - acmd, clen, flags, glen
+            - *continuous/non-continuous valid
```
Comment (weicaiyang) [inline hw/ip/csrng/data/csrng_testplan.hjson:107]: ditto

File: hw/ip/csrng/doc/checklist.md [hunk 3]
```diff
@@ -179,25 +179,25 @@ Review        | [V2_CHECKLIST_SCOPED][]               | Done        |
 
  Type         | Item                                    | Resolution  | Note/Collaterals
 --------------|-----------------------------------------|-------------|------------------
-Documentation | [DESIGN_DELTAS_CAPTURED_V2][]           | Not Started |
-Documentation | [DV_DOC_COMPLETED][]                    | Not Started |
-Testbench     | [FUNCTIONAL_COVERAGE_IMPLEMENTED][]     | Not Started |
-Testbench     | [ALL_INTERFACES_EXERCISED][]            | Not Started |
+Documentation | [DESIGN_DELTAS_CAPTURED_V2][]           | Done        |
+Documentation | [DV_DOC_COMPLETED][]                    | Done        |
```
Comment (weicaiyang) [inline hw/ip/csrng/doc/checklist.md:183]: can you fix the format problem and broken links in section `Global types & methods`.
https://docs.opentitan.org/hw/ip/edn/doc/dv/index.html

File: hw/ip/csrng/data/csrng_testplan.hjson (hunk 1 above)
Comment (weicaiyang) [reply hw/ip/csrng/data/csrng_testplan.hjson:65]: perhaps use TODO with comments
> // TODO, need to implement it in sequence

File: hw/ip/csrng/data/csrng_testplan.hjson (hunk 2 above)
Comment (senelson7) [reply hw/ip/csrng/data/csrng_testplan.hjson]: Done

File: hw/ip/csrng/doc/checklist.md (hunk 3 above)
Comment (senelson7) [reply hw/ip/csrng/doc/checklist.md:183]: Done

File: hw/ip/csrng/data/csrng_testplan.hjson [hunk 4]
```diff
@@ -59,17 +61,20 @@
             Verify genbits generated as predicted.
             Verify fips bit is passed through properly.
             Verify ability to access registers based on otp, enables.
+            Verify AES_HALT.
+            // TODO: Implement in sequence or agent
```
Comment (weicaiyang) [inline hw/ip/csrng/data/csrng_testplan.hjson:65]: if this is using push_pull agent, we can remove this TODO

File: hw/ip/csrng/doc/checklist.md (hunk 3 above)
Comment (weicaiyang) [reply hw/ip/csrng/doc/checklist.md:183]: the doc seems not fixed yet. Is the fix in a pending PR?

File: hw/ip/csrng/data/csrng_testplan.hjson [hunk 5]
```diff
@@ -83,6 +88,28 @@
...
+    }
+    {
+      name: csrng_cmds_cg
+      desc: '''
+            Covers that all csrng commands and variations have been tested for all apps.
+            Individual commands and command options that will be covered include:
+            - app
+            - acmd, clen, flags, glen
+            Crosses of app/acmd and acmd/clen, acmd/flags, acmd/glen
+            // TODO: Add following coverage
```
Comment (weicaiyang) [inline hw/ip/csrng/data/csrng_testplan.hjson:109]: ditto

File: hw/ip/csrng/doc/checklist.md (hunk 3 above)
Comment (senelson7) [reply hw/ip/csrng/doc/checklist.md:183]: No, this one. I fixed one issue by adding an example of a parameter to the "Global types and Methods" section. @weicaiyang Is there something else?

**Comment by senelson7** [comment]
@weicaiyang this PR is similar to the one you just approved for edn PR #11552. Should I remove the checklist from this one too? Then you'll approve?

**Comment by weicaiyang** [comment]
> @weicaiyang this PR is similar to the one you just approved for edn PR #11552. Should I remove the checklist from this one too? Then you'll approve?

yes, please split the checklist update into another PR. Then we can merge this one. Thanks

**Comment by senelson7** [comment]
Checklist removed

**Pull request merged**